lib/api/controllers/documentController.js

```diff
@@ -224,22 +223,24 @@ class DocumentController extends BaseController {
         modifiedRequest = newRequest;
 
         // @deprecated - Notifications before action will be removed in Kuzzle v2
-        this.kuzzle.notifier.publish(modifiedRequest, 'unknown', 'pending');
-
-        return this.engine.createOrReplace(modifiedRequest);
+        return this.kuzzle.notifier.publish(
+          modifiedRequest, 'unknown', 'pending');
       })
-      .then(response => {
-        this.kuzzle.indexCache.add(modifiedRequest.input.resource.index, modifiedRequest.input.resource.collection);
-
+      .then(() => this.engine.createOrReplace(modifiedRequest))
+      .then(_response => {
+        response = _response;
+        return this.kuzzle.indexCache.add(
```

Comment:
I didn't pay attention to that line, since it was already there. But now that you mention it, I can't find why this would be useful.

That line was added by a [3-years old PR](https://github.com/kuzzleio/kuzzle/pull/137/files#diff-d1355b0c631259c190837285da4f9880R41) and my best guess is that, at that time, it was useful to put in cache the indexes/collections created directly in ES while Kuzzle was running to prevent false negatives.

This is not useful anymore since the index cache now features a `hotreload` mechanism.

So I'll remove it. Good catch :+1: